api/common/src/system.rs

```diff
@@ -0,0 +1,23 @@
+//! Defines types for registering systems and event handlers.
+
+use crate::event::{EventId, OpaqueEvent};
+use crate::states::GameState;
+
+/// Specifies a description of a system.
+#[repr(C)]
+pub struct SystemSpec {
+    /// A pointer to the function to run when the system is invoked.
+    pub f: fn(*mut GameState),
```

Comment:
Can we change the name to something more descriptive than `f`, like `run`, `invoke`, or `step`?